pkg/model/estimator/local/regressor/model_weights.go

```diff
@@ -122,3 +125,24 @@ func (w ComponentModelWeights) String() string {
 	}
 	return fmt.Sprintf("%s (package: %v (core: %v, uncore: %v), dram: %v)", w.ModelName, w.Package, w.Core, w.Uncore, w.DRAM)
 }
+
+func (w ComponentModelWeights) Trainer() string {
+	if w.ModelName == "" {
+		return ""
+	}
+	modelNameSplits := strings.Split(w.ModelName, "_")
+	splitTrainer := strings.Join(modelNameSplits[0:len(modelNameSplits)-1], "_")
+	if isSupportedTrainer(splitTrainer) {
+		return splitTrainer
+	}
+	return ""
+}
```

Comment:
IMHO, models should have the trainer-name metadata embedded in the json itself so that this isn't needed.

Reply:
model_name has more complete information since it also embed the node_type and linked to model db.
I think we can add trainer_name additionally if needed.